src/InvoiceGenerator.BusinessLogic/ClientService.cs

```diff
@@ -0,0 +1,62 @@
+﻿using InvoiceGenerator.Domain;
+using InvoiceGenerator.Domain.Models;
+using InvoiceGenerator.ViewModels;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace InvoiceGenerator.BusinessLogic
+{
+    public class ClientService : IClientService
+    {
+        private readonly InvoiceDbContext _invoiceDbContext;
+
+        public ClientService(InvoiceDbContext invoiceDbContext)
+        {
+            _invoiceDbContext = invoiceDbContext;
+        }
+
+        public List<ClientViewModel> GetClients()
+        {
+            var clients = _invoiceDbContext.Clients.ToList(); // Assign a bunch of client objects to a list
+
+            var viewModels = new List<ClientViewModel>();
+
+            foreach (var db in clients)
+            {
+                viewModels.Add(ClientViewModel.FromDbModel(db)); // Add the client details to a list
+            }
+            return viewModels;
+        }
+
+        public List<ClientNameViewModel> GetClientNames()
+        {
+
+            var clientNames = _invoiceDbContext.Clients.ToList();
+
+            var nameModels = new List<ClientNameViewModel>();
+
+            foreach (var db in clientNames)
+            {
+                nameModels.Add(ClientNameViewModel.FromDbModel(db));
+            }
+            return nameModels;
+        }
+
+        public void AddClients(string clientName, string clientAddress, string contactName, string contactEmail)
```

Comment:
🌱 Might be worth taking in a Client View Model here, then creating a mapper method similar to the call to `nameModels.Add(ClientNameViewModel.FromDbModel(db))` which takes the ViewModel and produces a database model.